src/java/com/unifina/signalpath/text/DecodeByteArrayToString.java

```diff
@@ -0,0 +1,63 @@
+package com.unifina.signalpath.text;
+
+import com.unifina.signalpath.*;
+import javax.xml.bind.DatatypeConverter;
+import java.util.*;
+
+public class DecodeByteArrayToString extends AbstractSignalPathModule {
+
+	private ByteArrayInput in = new ByteArrayInput(this, "in");
+	private StringOutput out = new StringOutput(this, "out");
+	private StringOutput error = new StringOutput(this, "error");
+	private DecodeParameter decodeType = new DecodeParameter(this, "decodeType", DecodeParameter.HEX);
+
+	@Override
+	public void init() {
+		addInput(in);
+		addInput(decodeType);
+		addOutput(out);
+		addOutput(error);
+		in.setCanConnect(true);
+		decodeType.setCanConnect(false);
+		decodeType.setUpdateOnChange(true);
+	}
+
+	@Override
+	public void sendOutput() {
+
+		try {
+			switch (decodeType.getValue()){
+				case DecodeParameter.BASE64: out.send(DatatypeConverter.printBase64Binary(in.getValue()));
+					break;
+				default:
+				case DecodeParameter.HEX: out.send(DatatypeConverter.printHexBinary(in.getValue()));
+					break;
+			}
+		} catch (Exception e ){
+			error.send("Failed to parse byte array");
```

Comment:
Can a more specific Exception be caught? If not, would be good idea to log error, i.e.,

`private static final Logger log = Logger.getLogger(DecodeByteArrayToString.class);`

and then

`log.error(e);`